biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceBuilder.java

```diff
@@ -638,15 +646,15 @@ public void addRequirements(List<Requirement> requires) {
 		return capabilities;
 	}
 
-	public Map<Capability, Capability> from(Resource bundle) {
-		Map<Capability, Capability> mapping = new HashMap<>();
+	public Map<Capability, CapReqBuilder> from(Resource bundle) {
```

Comment:
Actually, upon further thinking, this change is, from a practical point of view, a binary incompatible change. Any call site getting a value from a map entry will have a compiler generated cast to `Capability`. So existing binaries doing something like:

```java
Capability cap = from(bundle).get(someCap);
```

will result in a ClassCastException at runtime since the compiler expects the value to be of type `Capability` and generates a cast to verify this.

So this sort of change is, in effect, binary incompatible.